cfme/tests/cloud_infra_common/test_provisioning.py

```diff
@@ -169,63 +168,55 @@ def test_provision_approval(appliance, provider, vm_name, smtp_test, request,
     # template, host, datastore = map(provisioning.get, ('template', 'host', 'datastore'))
 
     # It will provision two of them
+    # All the subject checks are normalized, because of newlines and capitalization
+
     vm_names = [vm_name + "001", vm_name + "002"]
-    if BZ(1628240).blocks and provider.one_of(CloudProvider):
+    if provider.one_of(CloudProvider):
         requester = ""
+        vm_type = "instance"
     else:
-        requester = "vm_provision@cfmeqe.com "
+        requester = "vm_provision@cfmeqe.com "  # include trailing space for clean formatting
+        vm_type = "virtual machine"
     collection = appliance.provider_based_collection(provider)
-    inst_args = {'catalog': {
-        'vm_name': vm_name,
-        'num_vms': '2'
-    }}
+    inst_args = {
+        'catalog': {
+            'vm_name': vm_name,
+            'num_vms': '2'
+        }
+    }
 
     vm = collection.create(vm_name, provider, form_values=inst_args, wait=False)
     try:
-        if provider.one_of(CloudProvider):
-            vm_type = "instance"
-        else:
-            vm_type = "virtual machine"
-
-        subject = VersionPicker({
-            LOWEST: "your request for a new vms was not autoapproved",
-            "5.10": "your {} request is pending".format(vm_type)
-        }).pick()
-        wait_for(
-            lambda:
-            len(filter(
-                lambda mail:
-                normalize_text(subject) in normalize_text(mail["subject"]),
-                smtp_test.get_emails())) == 1,
-            num_sec=90, delay=5)
-        subject = VersionPicker({
-            LOWEST: "virtual machine request was not approved",
-            "5.10": "{} request from {}pending approval".format(vm_type, requester)
-        }).pick()
-
-        wait_for(
-            lambda:
-            len(filter(
-                lambda mail:
-                normalize_text(subject) in normalize_text(mail["subject"]),
-                smtp_test.get_emails())) == 1,
-            num_sec=90, delay=5)
+        wait_for(lambda: len(smtp_test.get_emails()) >= 2,
+                 num_sec=90,
+                 delay=3)
     except TimedOutError:
-        subjects = ",".join([normalize_text(m["subject"]) for m in smtp_test.get_emails()])
-        logger.error("expected: %s, got emails: %s", subject, subjects)
-        raise
+        pytest.fail('Did not receive at least 2 emails from provisioning request, received: {}'
+                    .format(smtp_test.get_emails()))
+
+    expected_subject = normalize_text("your {} request is pending".format(vm_type))
+    from_subject = normalize_text("{} request from {}pending approval".format(vm_type, requester))
```

Comment:
Space between `{}` and `pending`

Reply:
The space should be included in the variable formatted into the string - IIRC 5.10 does not include the requester email

Reply:
it is present in requester :). Could we just move that space from requested to here ?

Reply:
Its dependent on cloud vs infra, and I would rather set it there instead of having a separate if statement where these are needed.